AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/security/control/PermissionService.java

```diff
@@ -850,6 +850,23 @@ public boolean hasPermissionToDelegatePermission(Permission permission, Resource
         return false;
     }
 
+    /**
+     * Checks if the caller already has a Restriction similar to the given Restriction
+     * Returns true if he already has a similar, equal or less restrictive Restriction
+     * Returns false if he does not have a similar or just one that is more restrictive than the given one
+     *
+     * @param newRestriction
+     */
+    public boolean callerHasIdenticalOrMoreGeneralRestriction(RestrictionEntity newRestriction) {
```

Comment:
Why can't `public boolean identicalOrMoreGeneralRestrictionExists(RestrictionEntity newRestriction)` be used here?
The checks should be the same?
Same for `createAutoAssignedRestriction`.
Also one method is called `createSelfAssignedRestrictions` and the other `createAutoAssignedRestriction` -> stick to either self or auto Please.

Reply:
Not really, as the system does not know which Role/s a User has. `identicalOrMoreGeneralRestrictionExists` can only check if a similar Restriction of a User OR a Role exists. But the system knows which Role the logged in User (caller) has, so it can check if a User already has a similar Restriction, no matter if from one of its Roles or directly as UserRestricion.